lib/Beam/Emitter.pm

```diff
@@ -15,20 +15,27 @@ use Beam::Listener;
 # The event listeners on this object, a hashref of arrayrefs of
 # EVENT_NAME => [ Beam::Listener object, ... ]
 
-=attr listeners
-
-An array containing the listeners which have subscribed to this
-emitter.  The array elements are instances of classes which inherit
-from L<Beam::Listener>
-
-=cut
-
-has listeners => (
+has _listeners => (
     is      => 'ro',
     isa     => HashRef,
     default => sub { {} },
 );
 
+=method listeners( event_name )
+
+Returns an array containing the listeners which have subscribed to the
+specified event from this emitter.  The array elements are instances
+of sub-classes of L<Beam::Listener>.
+
+=cut
+
+sub listeners {
+
+    my ( $self, $name ) = @_;
+
+    return $self->_listeners->{$name} || [];
```

Comment:
Another school of thought is to return arrayrefs rather than lists in order to reduce stack size when there's the possibility of returning a large number of items.  The returned arrayref is a new one, e.g. [ @{$arrayref} ].